server/openejb-ejbd/src/main/java/org/apache/openejb/server/ejbd/EjbRequestHandler.java

```diff
@@ -103,11 +103,17 @@ public Response processRequest(final ObjectInputStream in, final ProtocolMetaDat
         final SecurityService securityService = SystemInstance.get().getComponent(SecurityService.class);
         boolean failed = false;
         final CallContext call;
+        Object oldClientIdentity = null;
+
 
         try {
             try {
                 final Object clientIdentity = req.getClientIdentity();
                 if (clientIdentity != null) {//noinspection unchecked
+                    if (securityService.getCallerPrincipal() != null) {
```

Comment:
Not exactly sure why this is needed. When debugging securityService.disassociate() comes as null. Can you clarify why it was added?